src/lib/data/storage/tests/test_client.py

```
@@ -0,0 +1,1021 @@
+    """
+    Tests dispatch behavior for ``Client.upload_with_worker_inputs`` for
+    both list and generator source variants.
+    """
+
+    def setUp(self):
+        self.client = _make_client()
+
+    def test_upload_with_worker_inputs_list_passes_list_param(self):
+        """list source -> upload_worker_inputs is set on the params."""
+        worker_input = uploading.UploadWorkerInput(
+            source='/tmp/source',
```

Ruff (0.15.14)

[error] Probable insecure usage of temporary file or directory: "/tmp/source"

(S108)